core/src/upickle/core/Types.scala

```diff
@@ -277,16 +277,16 @@ trait Types{ types =>
         ctx.visitEnd(-1)
       }
     }
-    protected def writeSnippet[R, V](objectAttributeKeyWriteMap: CharSequence => CharSequence,
-                                     ctx: _root_.upickle.core.ObjVisitor[_, R],
-                                     mappedArgsI: String,
-                                     w: Writer[V],
-                                     value: V) = {
+    def writeSnippet[R, V](objectAttributeKeyWriteMap: CharSequence => CharSequence,
+                           ctx: _root_.upickle.core.ObjVisitor[_, R],
+                           mappedArgsI: String,
+                           w: Any,
+                           value: Any) = {
```

Comment:
Had to make these un-typed because I couldn't figure out how to make the Scala 3 macro quotes/splices play nicely with the path dependent `prefix.Writer` type. A bit gross but shouldn't affect perf or correctness